### PR TITLE
What is your nhs number

### PR DESCRIPTION
**Pull request by huwd**
# Screenshots
image

image

image

image

image

Also tested with my own NHS number (which i won't print here, it passes!)

### CODE REVIEW

File: app/controllers/coronavirus_form/nhs_number_controller.rb [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+
+  def clean_nhs_number(unclean_nhs_number)
+    unclean_nhs_number&.gsub(" ", "")
+  end
+
+  def validate_fields(nhs_number)
+    validation_response = validate_nhs_number_correctness(nhs_number)
+    return [] if validation_response[:valid]
+
+    puts validation_response.inspect
```
Comment (bevanloon) [inline app/controllers/coronavirus_form/nhs_number_controller.rb]: Should we remove this?

Comment (issyl0) [reply app/controllers/coronavirus_form/nhs_number_controller.rb]: Yeah, this is nowhere near ready yet, we'll tidy up later.

File: app/helpers/nhs_number_validator_helper.rb [hunk 2]
```diff
@@ -0,0 +1,68 @@
+# frozen_string_literal: true
+
+module NhsNumberValidatorHelper
+  def validate_nhs_number_correctness(number)
+    puts number.inspect
+    if number.nil?
```
Comment (bevanloon) [inline app/helpers/nhs_number_validator_helper.rb]: I know you've cleaned the number up above, but what about `.blank?` that will validate the empty string or a nil string or something like `"  "` if I remember rightly

Comment (huwd) [reply app/helpers/nhs_number_validator_helper.rb]: ah, good shout, and we want the empty string to return too right?

File: app/controllers/coronavirus_form/nhs_number_controller.rb [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+        text: validation_response[:message],
+      },
+    ]
+  end
+
+  PAGE = "nhs_number"
+  NEXT_PAGE = "check_answers" # TODO change to address
+
+  def previous_path
+    "/" # TODO
```
Comment (issyl0) [inline app/controllers/coronavirus_form/nhs_number_controller.rb]:
```suggestion
    coronavirus_form_know_nhs_number_path
```

File: app/controllers/coronavirus_form/nhs_number_controller.rb [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+    [
+      {
+        field: "nhs_number",
+        text: validation_response[:message],
+      },
+    ]
+  end
+
+  PAGE = "nhs_number"
+  NEXT_PAGE = "check_answers" # TODO change to address
```
Comment (issyl0) [inline app/controllers/coronavirus_form/nhs_number_controller.rb]:
```suggestion
  NEXT_PAGE = "essential_supplies"
```

**Review by issyl0 — CHANGES_REQUESTED** [review]
While we're here we may as well do the back/forwards routing.

File: app/controllers/coronavirus_form/nhs_number_controller.rb (hunk 1 above)
Comment (issyl0) [reply app/controllers/coronavirus_form/nhs_number_controller.rb]: It's gone now.

**Review by issyl0 — APPROVED** [review]

File: config/locales/en.yml [hunk 5]
```diff
@@ -178,9 +178,20 @@ en:
           label: Postcode
       basic_care_needs:
         title: Are your basic care needs being met at the moment?
-        hint: For example, washing and bathing yourself and keeping your house clean, with the help of friends or family.
+        hint: For example, washing and bathing yourself and keeping your house clean, with the help of friends or family.
```
Comment (issyl0) [inline config/locales/en.yml]:
```suggestion
        hint: For example, washing and bathing yourself and keeping your house clean, with the help of friends or family.
```

**Review by issyl0 — APPROVED** [review]
LGTM. 🚢

**Pull request merged**